app/src/main/java/com/example/poste/models/Post.java

```diff
@@ -67,6 +76,24 @@ public String getUrl() {
      */
     public String getId() { return id; }
 
+    /**
+     * Deletes this post from the data model and backend
+     */
+    public void deletePost(Folder parentFolder) {
+        Post delPost = this;
+        MyApiService apiService = RetrofitClient.getRetrofitInstance().create(MyApiService.class);
+        Call<ResponseBody> call = apiService.deletePost(new DeletePost(delPost.getId()));
+        call.enqueue(new Callback<ResponseBody>() {
+            @Override
+            public void onResponse(Call<ResponseBody> call, Response<ResponseBody> response) {
+                parentFolder.getPosts().remove(delPost);
+            }
+            @Override
+            public void onFailure(Call<ResponseBody> call, Throwable t) {
```

Comment:
As above, but this indicates a failure to send the API request at all (connectivity, etc)